src/status_im/ui2/screens/chat/components/chat_item.cljs

```diff
@@ -0,0 +1,199 @@
+       [preview-label :t/community-message-preview {:community-name name}]))])
+
+(def memo-timestamp
+  (memoize
+   (fn [timestamp]
+     (string/upper-case (time/to-short-str timestamp)))))
+
+(defn unviewed-indicator [{:keys [unviewed-mentions-count
+                                  unviewed-messages-count
+                                  public?]}]
+  (when (pos? unviewed-messages-count)
+    [rn/view {:position :absolute :right 16}
+     (cond
+       (and public? (not (pos? unviewed-mentions-count)))
+       [rn/view {:style                  styles/public-unread
+                 :accessibility-label :unviewed-messages-public}]
+
+       (and public? (pos? unviewed-mentions-count))
+       [badge/message-counter unviewed-mentions-count]
+
+       :else
+       [badge/message-counter unviewed-messages-count])]))
+
+(defn icon-style []
+  {:color           colors/black
+   :width           15
+   :height          15
+   :container-style {:top          13 :left 72
+                     :position     :absolute
+                     :width        15
+                     :height       15
+                     :margin-right 2}})
+
+(defn chat-item-icon [muted private-group? public-group?]
+  (cond
+    muted
+    [icons/icon :main-icons/tiny-muted (assoc (icon-style) :color colors/gray)]
+    private-group?
+    [icons/icon :main-icons/tiny-group (icon-style)]
+    public-group?
+    [icons/icon :main-icons/tiny-public (icon-style)]
+    :else
+    [icons/icon :main-icons/tiny-new-contact (icon-style)]))
+
+(defn chat-item-title [chat-id muted group-chat chat-name edit?]
+  [quo2.text/text {:weight              :semi-bold
+                   :color               (when muted :secondary)
+                   :accessibility-label :chat-name-text
+                   :ellipsize-mode      :tail
+                   :number-of-lines     1
+                   :style               {:position :absolute
+                                         :left     72
+                                         :top      10
+                                         :right    (if edit? 50 90)}}
+   (if group-chat
+     (utils/truncate-str chat-name 30)
+     ;; This looks a bit odd, but I would like only to subscribe
+     ;; if it's a one-to-one. If wrapped in a component styling
+     ;; won't be applied correctly.
+     (first @(re-frame/subscribe [:contacts/contact-two-names-by-identity chat-id])))])
+
+(defn chat-list-item [home-item opts]
+  (let [{:keys [chat-id chat-name color group-chat muted emoji highlight edit? public? unviewed-messages-count last-message]} home-item
+        background-color (when highlight (colors/get-color :interactive-02))]
+    [rn/touchable-opacity (merge {:style {:height 64 :background-color background-color}} opts)
+     [:<>
+      (when (pos? unviewed-messages-count)
+        [rn/view {:position :absolute :top 2 :left 8 :right 8 :bottom 2 :border-radius 16 :background-color quo2.colors/primary-50-opa-5}])
```

Comment:
[Code style] Just personal preference here, but I think this line is a tad too unreadable. How about breaking the lines for each pair?